operators/src/upgrade/helm/client.rs

```diff
@@ -0,0 +1,249 @@
+
+        // Check that the version command gives a version.
+        if !out_str.starts_with("v3.") {
+            return Err(Error::HelmVersionNotFound(
+                "Helm version 3 not installed, installed version:{}".to_string(),
+                out_str,
+            ));
+        }
+
+        // If checks succeed, create Helm client.
+        Ok(Self {
+            args: HelmArgs::default(),
+            version: Some(out_str),
+            chart: Chart::default(),
+        })
+    }
+
+    /// Get version.
+    pub(crate) fn version(&self) -> &String {
+        self.version.as_ref().unwrap()
+    }
+
+    /// Set chart in helm client.
+    pub(crate) fn set_chart(mut self, name: String, namespace: String) -> Result<Self, Error> {
+        match Chart::get_installed_chart_by_name(name, namespace) {
+            Ok(chart) => {
+                self.chart = chart;
+            }
+            Err(err) => {
+                return Err(err);
+            }
+        }
+        Ok(self)
+    }
+
+    /// Set arguments for the helm commands.
+    fn set_args(&mut self, values: Option<Vec<PathBuf>>, opts: Option<Vec<(String, String)>>) {
+        if !self.chart.namespace().is_empty() {
+            self.args.set_namespace(self.chart.namespace().to_string());
+        }
+        if !self.chart.name().is_empty() {
+            self.args.set_name(self.chart.name().to_string());
+        }
+        if let Some(values) = values {
+            self.args.set_values(values);
+        }
+        if let Some(opts) = opts {
+            self.args.set_opts(opts);
+        }
+    }
+
+    /// Apply arguments for helm command.
+    fn apply_args(&self, command: &mut Command) {
+        command
+            .arg(self.args.get_name())
+            .arg("--namespace")
+            .arg(self.args.get_namespace());
+
+        for value_path in self.args.get_values() {
+            command.arg("--values").arg(value_path);
+        }
+
+        for (key, val) in self.args.get_opts() {
+            command.arg("--set").arg(format!("{}={}", key, val));
+        }
+    }
+
+    /// Helm upgrade command.
+    pub(crate) fn upgrade(
+        &mut self,
+        values: Vec<PathBuf>,
+        opts: Vec<(String, String)>,
+    ) -> Result<(), Error> {
+        self.set_args(Some(values), Some(opts));
+
+        self.run_command(vec!["upgrade".to_string()])?;
+        Ok(())
+    }
+
+    /// Command to get values of the installed chart.
+    pub(crate) fn get_values(&mut self) -> Result<(), Error> {
+        self.set_args(None, None);
+        let output = self.run_command(vec!["get".to_string(), "values".to_string()])?;
+
+        let x = String::from_utf8(output.stdout)
+            .map_err(|_| Error::Utf8Error("Unable to parse values into string".to_string()))?;
+        println!("{:?}", x);
+
+        Ok(())
+    }
+
+    /// Run helm commands.
+    fn run_command(&self, command_type_args: Vec<String>) -> Result<Output, Error> {
```

Comment:
Isn't it a bit confusing that we have `command_type_args` here, but within we add more `args` ?
What about using a builder like patter, to build a helm command on demand explicitly with types, example:

```rust
impl HelmClient { 
  fn args(&self) -> HelmArgs { self.default_args.clone() }
}
impl HelmArgs { fn with_xxx(self) -> Self {} fn run(self) -> Result<O,E> {} ... }
let client = HelmClient ...

/// this way it's explicit that we have an args
client.args().with_cmd(vec!["this", "that"]).run()?;
/// and easy to override options, example:
client.args().with_namespace("bob").with_cmd(vec!["this", "that"]).run()?;
```